extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java

```diff
@@ -126,14 +139,22 @@ public void onLightStateChanged(HueBridge bridge, FullLight light) {
 
     private ThingUID getThingUID(FullLight light) {
         ThingUID bridgeUID = hueBridgeHandler.getThing().getUID();
-        ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, light.getModelID().replaceAll("[^a-zA-Z0-9_]", "_"));
+        ThingTypeUID thingTypeUID = getThingTypeUID(light);
 
-        if (getSupportedThingTypes().contains(thingTypeUID)) {
+        if (thingTypeUID != null && getSupportedThingTypes().contains(thingTypeUID)) {
             String thingLightId = light.getId();
-            ThingUID thingUID = new ThingUID(thingTypeUID, bridgeUID, thingLightId);
+            ThingTypeUID modelUID = new ThingTypeUID(BINDING_ID,
+                    light.getModelID().replaceAll(HueLightHandler.NORMALIZE_ID_REGEX, "_"));
+            ThingUID thingUID = new ThingUID(modelUID, bridgeUID, thingLightId);
```

Comment:
In general, you are right. We could leave out the thing type id when constructing a thing UID. I even tried it but there are far too many places in the code, where implicitly the thing type id is retrieved from the thing UID. If I would like to change this, it would be far beyond the scope of this PR. As it is ugly and the methods dealing with thing type id as part of a thing UID are already deprecated I would suggest we create an issue for it to fix it in another PR.